src/main/java/com/twilio/rest/api/v2010/account/usage/record/AllTimeReader.java

```diff
@@ -201,16 +172,12 @@ private void addQueryParams(final Request request) {
             request.addQueryParam("Category", category.toString());
         }
         
-        if (absoluteStartDate != null) {
-            request.addQueryParam("StartDate", absoluteStartDate.toString(Request.QUERY_STRING_DATE_FORMAT));
-        } else if (rangeStartDate != null) {
-            request.addQueryDateRange("StartDate", rangeStartDate);
+        if (startDate != null) {
+            request.addQueryParam("StartDate", DateConverter.dateStringFromLocalDate(startDate));
```

Comment:
If we were to go that route, I feel we would have to add `addQueryParam` methods for each type of query parameter that we have. This would case a lot of clutter in `Request`. I'd also be concerned about running into a case where we have a generic type that conflicts and we are unable to perfectly overload methods.